src/sage/schemes/elliptic_curves/addition_formulas_ring.py

```diff
@@ -0,0 +1,89 @@
+
+def add(E, P, Q):
```

Comment:
```suggestion
def _add(E, P, Q):
```
Internal functions should start with an underscore.